WIP - 110 Change board name

RikTheunis opened the pull request:
Tackle issue #110. Add functionality to change the name of the board.

- [x] Add functionality to change the name of the board.
- [x] Add a test for the above functionality.
- [x] Add pencil icon on hover.
- [x] Change layout to project maintainers' liking. 
- [x] Fix flickering pencil.

RikTheunis commented:
Any feedback @Roshanjossey ?

Roshanjossey commented:
@RikTheunis:Great job. Lemme try this locally. I'll give you my feedback in within 2 hours.

Roshanjossey commented:
@RikTheunis: I'm happy with the feature.

One nice to have addition would be to show an edit icon (pencil icon) on hover.

@glauberramos Could you take a look at it?

Roshanjossey commented:
@RikTheunis, Just tried out your change. Looks wonderful.

We'll wait for @glauberramos to share his thoughts about this. I'm super happy with your changes

RikTheunis commented:
@Roshanjossey Thanks! I appreciate the quick responses!

glauberramos commented:
@Roshanjossey, @RikTheunis  Hey guys just got back from vacation, I will look into this later today. :D thanks for the good work

glauberramos commented:
Hey @RikTheunis I really liked the feature, it is working perfectly. I just have some minor feedbacks on the interface.

The cancel link should be aligned vertically, the save button as well, and also give "margin-right: 5px" to the save button, should look like this:

![captura de tela 2017-03-08 as 23 18 56](https://cloud.githubusercontent.com/assets/559434/23733095/b6385e98-0455-11e7-8c0f-f91aa219bd42.png)

Also I am not sure we need to pencil on hover, it makes it not consistent with the column name, because column name does not have the pencil on hover. I think we can remove it for now.

Thanks for the great work!

RikTheunis commented:
Thanks, @glauberramos. I totally agree about the layout, I  will work on that. Though I personally think we should have pencils on hover for both the name and the columns, because it improves the usability, but I will remove it for now.

glauberramos commented:
@RikTheunis I can add an issue and then you can work on that next, what do you think? Or do you prefer doing it all here?

Roshanjossey commented:
@glauberramos: Just wanna make sure we're on the same page here.

The new issue to be created is about `edit-pencil` in all the places that needs similar editing right?

If that is the case, Lets keep the pencils in board name and add more pencils later.

glauberramos commented:
Ok, I think we can keep the pencil, but we need to fix an issue with it, right now when you click to edit, it is doing a flickering thing, the pencil shows on the left and then dessapears. @RikTheunis can we fix it?

Check the pencil showing on the left:
![gif](https://cloud.githubusercontent.com/assets/559434/23767299/e71c2e18-04e6-11e7-89e8-fcadcb873f31.gif)

RikTheunis commented:
@glauberramos I will fix it! Thanks for the feedback guys!

RikTheunis commented:
Hey @Roshanjossey and @glauberramos, I changed the layout (see picture below) and fixed the flickering pencil. Please let me know if you have any feedback!

![distributed_change_board_name](https://cloud.githubusercontent.com/assets/17219915/23827204/3faa2fc6-06ae-11e7-9399-66c5c69af0ac.png)

glauberramos commented on stylesheets/_base.scss:
```diff
@@ -144,6 +144,25 @@ ul {
   border: none;
 }
 
+.menu #board-name + .fa-pencil{
```
Instead of `.menu #board-name + .fa-pencil{` just use `#board-name + .fa-pencil {`

glauberramos replied:
Also it is missing a space before {

glauberramos commented on stylesheets/_base.scss:
```diff
@@ -144,6 +144,25 @@ ul {
+.menu .editing-board-name a.blue {
+  font-size: .9em;
+}
+
```
Same comment for all these lines, you don't need to use `.menu` and it is missing a space before {

glauberramos commented:
I am happy with the changes, merging now :dagger: 

Thanks everybody
